packages/edit-site/src/components/add-new-pattern/index.js

```diff
@@ -97,6 +98,31 @@ export default function AddNewPattern() {
 		title: __( 'Import pattern from JSON' ),
 	} );
 
+	const categoryMap = usePatternCategoriesMap();
+	async function createPatternCategory( existingTerm ) {
+		try {
+			// Since we have an existing core category we need to match the new user category to the
+			// correct slug rather than autogenerating it to prevent duplicates, eg. the core `Headers`
+			// category uses the singular `header` as the slug.
+			const termData = {
+				name: existingTerm.label,
+				slug: existingTerm.name,
+			};
+			const newTerm = await saveEntityRecord(
+				'taxonomy',
+				'wp_pattern_category',
+				termData,
+				{ throwOnError: true }
+			);
+			invalidateResolution( 'getUserPatternCategories' );
+			return newTerm.id;
+		} catch ( error ) {
+			if ( error.code !== 'term_exists' ) {
+				throw error;
+			}
+			return error.data.term_id;
+		}
+	}
```

Comment:
Could we call the new hook `useAddPatternCategory` and have it return `createPatternCategory` and `categoryMap` so we don't have to replicate this method here?

Reply:
Sounds good. I updated the code.